src/Chapter3.hs

```diff
@@ -970,6 +1050,26 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Day
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday deriving (Enum)
+isWeekend:: Day -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: Day -> Day
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: Day -> Int
+daysToParty Friday = 0
+daysToParty d = 1 + daysToParty (nextDay d) 
```

Comment:
Interesting recursive solution!
Though you can try to implement it with the `Enum` instance 🙂